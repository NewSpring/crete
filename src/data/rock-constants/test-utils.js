/* eslint-disable import/prefer-default-export */
export const buildGetMock = (response, dataSource) => {
  const get = jest.fn();
  if (Array.isArray(response) && Array.isArray(response[0])) {
    response.forEach((responseVal) => {
      get.mockReturnValueOnce(
        new Promise((resolve) => resolve(dataSource.normalize(responseVal)))
      );
    });
  }
  get.mockReturnValue(
    new Promise((resolve) => resolve(dataSource.normalize(response)))
  );
  return get;
};
