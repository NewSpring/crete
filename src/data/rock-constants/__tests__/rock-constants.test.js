import { fetch } from 'apollo-server-env';
import ApollosConfig from '@apollosproject/config';
import { dataSource as RockConstants } from '../index';
import { buildGetMock } from '../test-utils';

ApollosConfig.loadJs({
  ROCK_MAPPINGS: {
    INTERACTIONS: {
      CHANNEL_NAME: 'Apollos App',
      COMPONENT_NAME: 'Apollos Content Item',
      CHANNEL_MEDIUM_TYPE_ID: 512,
    },
    ENTITY_TYPES: {
      ApollosGroup: 'Group',
    },
  },
});

describe('RockConstants', () => {
  beforeEach(() => {
    fetch.resetMocks();
  });
  it("creates a content item Component if it doesn't exist", async () => {
    const dataSource = new RockConstants();
    dataSource.modelType = buildGetMock(
      { id: 101, friendlyName: 'Content Channel Item' },
      dataSource
    );
    dataSource.get = buildGetMock([[], { Id: 1 }], dataSource);
    dataSource.post = buildGetMock('1', dataSource);
    const result = await dataSource.contentItemInteractionComponent({
      contentItemId: 7,
      contentTitle: 'Some Title',
    });
    expect(result).toMatchSnapshot();
    expect(dataSource.modelType.mock.calls).toMatchSnapshot();
    expect(dataSource.get.mock.calls).toMatchSnapshot();
    expect(dataSource.post.mock.calls).toMatchSnapshot();
  });
  it('finds the content item Component if it exists', async () => {
    const dataSource = new RockConstants();
    dataSource.get = buildGetMock([{ Id: 1 }], dataSource);
    dataSource.post = jest.fn();
    dataSource.modelType = buildGetMock(
      { id: 101, friendlyName: 'Content Channel Item' },
      dataSource
    );
    const result = await dataSource.contentItemInteractionComponent({
      contentItemId: 7,
      contentTitle: 'Some Title',
    });
    expect(result).toMatchSnapshot();
    expect(dataSource.get.mock.calls).toMatchSnapshot();
    expect(dataSource.post.mock.calls.length).toBe(0);
  });
});
