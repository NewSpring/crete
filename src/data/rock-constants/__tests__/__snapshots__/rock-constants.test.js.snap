// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RockConstants creates a content item Component if it doesn't exist 1`] = `Array []`;

exports[`RockConstants creates a content item Component if it doesn't exist 2`] = `
Array [
  Array [
    "ContentItem",
  ],
]
`;

exports[`RockConstants creates a content item Component if it doesn't exist 3`] = `
Array [
  Array [
    "InteractionChannels?%24filter=%28Name%20eq%20%27Apollos%20App%27%29%20and%20%28UsesSession%20eq%20true%29%20and%20%28IsActive%20eq%20true%29%20and%20%28ComponentEntityTypeId%20eq%20101%29%20and%20%28ChannelTypeMediumValueId%20eq%20512%29",
    Object {},
    Object {
      "cacheOptions": Object {
        "ttl": 86400,
      },
    },
  ],
  Array [
    "/InteractionChannels/1",
  ],
  Array [
    "InteractionComponents?%24filter=%28Name%20eq%20%27Apollos%20Content%20Item%20-%207%27%29%20and%20%28InteractionChannelId%20eq%201%29%20and%20%28EntityId%20eq%207%29",
    Object {},
    Object {
      "cacheOptions": Object {
        "ttl": 86400,
      },
    },
  ],
]
`;

exports[`RockConstants creates a content item Component if it doesn't exist 4`] = `
Array [
  Array [
    "/InteractionChannels",
    Object {
      "ChannelTypeMediumValueId": 512,
      "ComponentEntityTypeId": 101,
      "IsActive": true,
      "Name": "Apollos App",
      "UsesSession": true,
    },
  ],
]
`;

exports[`RockConstants finds the content item Component if it exists 1`] = `
Object {
  "id": 1,
}
`;

exports[`RockConstants finds the content item Component if it exists 2`] = `
Array [
  Array [
    "InteractionChannels?%24filter=%28Name%20eq%20%27Apollos%20App%27%29%20and%20%28UsesSession%20eq%20true%29%20and%20%28IsActive%20eq%20true%29%20and%20%28ComponentEntityTypeId%20eq%20101%29%20and%20%28ChannelTypeMediumValueId%20eq%20512%29",
    Object {},
    Object {
      "cacheOptions": Object {
        "ttl": 86400,
      },
    },
  ],
  Array [
    "InteractionComponents?%24filter=%28Name%20eq%20%27Apollos%20Content%20Item%20-%207%27%29%20and%20%28InteractionChannelId%20eq%201%29%20and%20%28EntityId%20eq%207%29",
    Object {},
    Object {
      "cacheOptions": Object {
        "ttl": 86400,
      },
    },
  ],
]
`;
